.ci/test_r_package.sh

```diff
@@ -216,3 +216,17 @@ if [[ $OS_NAME == "macos" ]] && [[ $R_BUILD_TYPE == "cran" ]]; then
         exit -1
     fi
 fi
+
+# this check makes sure that no "warning: unknown pragma ignored" logs
+# reach the user leading them to believe that something went wrong
+if [[ $OS_NAME == "macos" || $OS_NAME == "linux" ]] && [[ $R_BUILD_TYPE == "cran" ]]; then
```

Comment:
I love this extra check in CI, thanks!

You can safely remove this first condition. This shell script will only be run on Linux and Mac builds.

We have a separate one for Windows: https://github.com/microsoft/LightGBM/blob/master/.ci/test_r_package_windows.ps1

```suggestion
if [[ $R_BUILD_TYPE == "cran" ]]; then
```